aggsender/flows/flow_aggchain_prover.go

```diff
@@ -357,6 +360,58 @@ func (a *AggchainProverFlow) UpdateAggchainData(
 	return nil
 }
 
+// VerifyCertificate verifies the new certificate
+// This function is used in the validator to verify the certificate
+func (a *AggchainProverFlow) VerifyCertificate(
+	ctx context.Context,
+	cert *agglayertypes.Certificate,
+	lastBlockInCert uint64,
+	lastSettledBlock uint64) error {
+	if cert.AggchainData == nil {
+		return fmt.Errorf("aggchainProverFlow: certificate AggchainData is nil")
+	}
+
+	aggchainDataProof, ok := cert.AggchainData.(*agglayertypes.AggchainDataProof)
+	if !ok {
+		// If the AggchainData is of type MultisigWithProof, we extract the proof and verify it
+		aggchainDataMultisigWithProof, ok := cert.AggchainData.(*agglayertypes.AggchainDataMultisigWithProof)
+		if !ok {
+			return fmt.Errorf("aggchainProverFlow: certificate AggchainData is of unknown type %T", cert.AggchainData)
+		}
+
+		aggchainDataProof = aggchainDataMultisigWithProof.AggchainProof
+	}
```

Comment:
IMO, it is cleaner this way

```suggestion
	var aggchainDataProof *agglayertypes.AggchainDataProof

	switch v := cert.AggchainData.(type) {
	case *agglayertypes.AggchainDataProof:
		aggchainDataProof = v
	case *agglayertypes.AggchainDataMultisigWithProof:
		aggchainDataProof = v.AggchainProof
	default:
		return fmt.Errorf("aggchainProverFlow: certificate AggchainData is of unknown type %T", cert.AggchainData)
	}
```

Reply:
Fix: https://github.com/agglayer/aggkit/pull/1060/commits/91fd403435cdc6ba89b32dd03edeec8265e20ce2